app/talk/tags.jsx

```diff
@@ -0,0 +1,189 @@
+import React from  'react';
+import PropTypes from  'prop-types';
+import { Link } from  'react-router';
+import talkClient from  'panoptes-client/lib/talk-client';
+import apiClient from  'panoptes-client/lib/api-client';
+import Paginator from  './lib/paginator';
+import getSubjectLocation from  '../lib/get-subject-location';
+import resourceCount from  './lib/resource-count';
+import ProjectLinker from  './lib/project-linker';
+import Loading from '../components/loading-indicator';
+
+import ActiveUsers from './active-users';
+import PopularTags from './popular-tags';
+import Thumbnail from '../components/thumbnail';
+
+export default class TalkTags extends React.Component {
+  constructor() {
+    super()
+
+    this.state = {
+      meta: {},
+      tags: null
+    }
+  }
+
+  componentDidMount() {
+    const { params, project } = this.props
+    const slugInURL = (params) ? `${params.owner}/${params.name}` : '';
+    const slugInProject = (project) ? project.slug : '';
+
+    if (slugInURL === slugInProject) {
+      this.getTags()
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const pageChanged = nextProps.location.query.page !== this.props.location.query.page
+    const differentTag = nextProps.params.tag !== this.props.params.tag
+    const differentProject = nextProps.project !== this.props.project
+
+    if (pageChanged || differentTag || differentProject) {
+      const nextPage = (differentTag) ? 1 : nextProps.location.query.page
+      this.getTags(nextPage, nextProps.params.tag, nextProps.project)
+    }
+  }
+
+  getTags(page = this.props.location.query.page, name = this.props.params.tag, project = this.props.project) {
+    page || (page = 1);
```

Comment:
Is this assigning page to 1 in the case it is undefined from `this.props.location.query.page`?